internal/monitors/elasticsearch/elasticsearch.go

```diff
@@ -0,0 +1,276 @@
+}
+
+// Monitor for conviva metrics
+type Monitor struct {
+	Output  types.Output
+	cancel  context.CancelFunc
+	ctx     context.Context
+	timeout time.Duration
+}
+
+func init() {
+	monitors.Register(monitorType, func() interface{} { return &Monitor{} }, &Config{})
+}
+
+// Configure monitor
+func (m *Monitor) Configure(conf *Config) error {
+	esClient := client.NewESClient(conf.Host, conf.Port, conf.UseHTTPS, conf.Username, conf.Password)
+	m.ctx, m.cancel = context.WithCancel(context.Background())
+	var nodeIsCurrentMaster bool
+	var defaultDimensions map[string]string
+
+	// Collect node stats
+	utils.RunOnInterval(m.ctx, func() {
+		// Collect info about master for the cluster
+		masterInfoOutput, err := esClient.GetMasterNodeInfo()
+
+		if err != nil {
+			logger.WithError(err).Errorf("Failed to GET master node info")
+			return
+		}
+
+		// Collect Node level stats
+		nodeStatsOutput, err := esClient.GetNodeAndThreadPoolStats()
+
+		if err != nil {
+			logger.WithError(err).Errorf("Failed to GET node stats")
+			return
+		}
+
+		nodeDimensions, err := prepareNodeMetricsDimensions(nodeStatsOutput.NodeStats)
+
+		if err != nil {
+			logger.WithError(err).Errorf("Failed to prepare node dimensions")
+			return
+		}
+
+		nodeIsCurrentMaster, err = isCurrentMaster(nodeDimensions["node_id"], masterInfoOutput)
+
+		if err != nil {
+			logger.WithError(err).Errorf("Failed to determine whether current node is the master")
+			return
+		}
+
+		clusterName := nodeStatsOutput.ClusterName
+		defaultDimensions, err = prepareDefaultDimensions(conf.Cluster, clusterName)
+
+		if err != nil {
+			logger.WithError(err).Errorf("Failed to prepare plugin_instance dimension")
+			return
+		}
+
+		nodeMetricsDims := utils.MergeStringMaps(defaultDimensions, nodeDimensions)
+
+		dps := client.GetNodeStatsDatapoints(nodeStatsOutput, nodeMetricsDims, utils.StringSliceToMap(conf.ThreadPools), utils.StringSliceToMap(conf.EnableEnhancedNodeStatsForIndexGroups), map[string]bool{
+			client.HTTPStatsGroup:       conf.EnableEnhancedHTTPStats,
+			client.JVMStatsGroup:        conf.EnableEnhancedJVMStats,
+			client.ProcessStatsGroup:    conf.EnableEnhancedProcessStats,
+			client.ThreadpoolStatsGroup: conf.EnableEnhancedThreadPoolStats,
+			client.TransportStatsGroup:  conf.EnableEnhancedTransportStats,
+		})
+
+		// If current node is not the master, collect only node level stats
+		if !nodeIsCurrentMaster {
+			m.sendDatapoints(dps)
+			return
+		}
+
+		// Collect cluster level stats
+		clusterStatsOutput, err := esClient.GetClusterStats()
+
+		if err != nil {
+			logger.WithError(err).Errorf("Failed to GET cluster stats")
+			return
+		}
+
+		dps = append(dps, client.GetClusterStatsDatapoints(clusterStatsOutput, defaultDimensions, conf.EnableEnhancedClusterHealthStats)...)
+
+		if !*conf.EnableIndexStats {
+			m.sendDatapoints(dps)
+			return
+		}
+
+		// Collect Index stats
+		if !conf.IndexStatsMasterOnly || nodeIsCurrentMaster {
```

Comment:
How will this ever be reached since you return on line 157 if the node isn't the current master?